contracts/truffle.js

```diff
@@ -1,5 +1,6 @@
 const HDWalletProvider = require("truffle-hdwallet-provider")
 const PrivateKeyProvider = require('truffle-privatekey-provider')
+const NonceTrackerSubprovider = require("web3-provider-engine/subproviders/nonce-tracker")
```

Comment:
Do we need to add web3-provider-engine as a dev dependency in package.json ? I think I had to install it before being able to run the truffle script with this change.

Reply:
Done.

Though `web3-provider-engine` is already a transitive dependency through `truffle-hdwallet-provider`, you're right that it's still good to have an explicit dependency.